### DISCUSSION TITLE
Suggestions for serializing individual nodes?

### DISCUSSION THREAD

**Discussion by haveanicedavid** [outline/rich-markdown-editor#569] [Q&A]
Hi! I'm trying to configure a custom `value()` output that requires serializing individual nodes to markdown (instead of the entire document). Say I wanted to serialize all paragraphs:

```ts
this.view.state.doc.content.forEach(node => {
  if (node.type.name === "paragraph") {
    const serialized = this.serializer.serialize(node);
  }
});
```

This will produce an output with correct text, but not include any marks ie if the paragraph is `Some _italic text_`, it will serialize to `Some italic text`.

I've been digging through the serializer code, but was hoping someone might have a suggestion as to what's going wrong

**Comment by tommoor** [comment]
Yea, I'm not sure – you're probably the first person to do this as the serializer is designed to work on a whole document. Perhaps you could create a fake "doc" node before passing into the serialize method? 😬
